### PR TITLE
Some missing language text keys 

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue #10798
#### Summary of Changes

Added  few strings but mainly it was correcting errors to make sure the correct string is bing used

### CODE REVIEW

File: libraries/joomla/language/stemmer.php [hunk 1]
```diff
@@ -55,7 +55,7 @@ public static function getInstance($adapter)
 		if (!class_exists($class))
 		{
 			// Throw invalid adapter exception.
-			throw new RuntimeException(JText::sprintf('JLIB_STEMMER_INVALID_STEMMER', $adapter));
+			throw new RuntimeException(JText::sprintf('COM_FINDER_INDEXER_INVALID_STEMMER', $adapter));
```
Comment (wilsonge) [inline libraries/joomla/language/stemmer.php]: We shouldn't be using finder strings in the core libraries

File: libraries/cms/table/corecontent.php [hunk 2]
```diff
@@ -95,7 +95,7 @@ public function check()
 	{
 		if (trim($this->core_title) == '')
 		{
-			$this->setError(JText::_('LIB_CMS_WARNING_PROVIDE_VALID_NAME'));
+			$this->setError(JText::_('COM_CONTENT_WARNING_PROVIDE_VALID_NAME'));
```
Comment (wilsonge) [inline libraries/cms/table/corecontent.php]: We shouldn't be using component strings in the core library folders

Comment (brianteeman) [inline libraries/cms/table/corecontent.php]: that means creating a new string - i as trying to save the TT from some work ;)
but i will create the new string if required

File: libraries/joomla/language/stemmer.php (hunk 1 above)
Comment (brianteeman) [inline libraries/joomla/language/stemmer.php]: that means creating a new string - i as trying to save the TT from some work ;)
but i will create the new string if required

File: libraries/cms/table/corecontent.php (hunk 2 above)
Comment (wilsonge) [inline libraries/cms/table/corecontent.php]: New strings for both please

Comment (brianteeman) [inline libraries/cms/table/corecontent.php]: I've reverted both of these changes but when looking to see where to create these strings I am confused as there are other component strings in core library folders https://github.com/joomla/joomla-cms/blob/staging/libraries/cms/table/contenttype.php#L73

so where should I be creating thee two strings?

Comment (infograf768) [inline libraries/cms/table/corecontent.php]: `LIB_CMS_WARNING_PROVIDE_VALID_NAME` should imho be corrected to
`JLIB_CMS_WARNING_PROVIDE_VALID_NAME` 
and placed in en-GB.lib_joomla.ini
as well for `JLIB_STEMMER_INVALID_STEMMER`

**Comment by infograf768** [comment]
Suggestion:
Change 
COM_CONTACT_SESSION_INVALID="Invalid session cookie. Please check that you have cookies enabled in your web browser."
to a JLIB_ENVIRONMENT_SESSION_INVALID string as It could be used in the future in other extensions than contacts.

It would be placed just after
JLIB_ENVIRONMENT_SESSION_EXPIRED="Your session has expired. Please log in again."

**Comment by brianteeman** [comment]
> COM_CONTACT_SESSION_INVALID
This is inside a function "Method to submit the contact form and send an email" so I dont think it is correct to make that change

**Comment by infograf768** [comment]
Many JLIB_ strings are used in specific extension methods. But nevermind

**Comment by wilsonge** [comment]
`JLIB_` strings belong in the library translation files but can be used by extensions
extension strings belong in the extension language file and should only be used by that extension.

**Comment by brianteeman** [comment]
I have made the requested changed for 
JLIB_STEMMER_INVALID_STEMMER
JLIB_CMS_WARNING_PROVIDE_VALID_NAME

Still not sure I understand the comments about COM_CONTACT_SESSION_INVALID should i move it to the JLIB_ or not?

**Comment by infograf768** [comment]
I suggested you change that constant to a JLIB_ one and create the string in lib.joomla.ini

**Comment by brianteeman** [comment]
OK thats done - please test and then we can finish with this one

**Comment by brianteeman** [comment]
@wilsonge @infograf768 all the changes you wanted have been done - please test so we can get it merged and give translators as much time as possible

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**